### PR TITLE
Only rotate server cert upon expiry

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

Change server cert rotation behavior. Before, it was done every time the `EnsureCertificate` method was called. Now certificate expiration date&time are taken into account.

- Introduce `serverCertGenerator` - takes care of rotation
- Introduce `certExpiryChecker` to be used by both ca and server certificate providers
- Introduce `certChainChecker` to be used by the server certificate provider
- Periodically check the certificates to make the change compatible with legacy reconciler-based installation (where no Telemetry CR is reconciled)

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/16626

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -67,8 +67,6 @@ import (
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that exec-entrypoint and run can make use of them.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
-	//nolint:gosec // pprof package is required for performance analysis.
-	_ "net/http/pprof"
```
Comment (chrkl) [inline main.go:71]: Re-add

File: internal/webhookcert/server_cert_provider.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+func (p *serverCertProviderImpl) provideCert(ctx context.Context, config serverCertConfig) ([]byte, []byte, error) {
+	var err error
+	var serverCertPEM, serverKeyPEM []byte
+	serverCertPEM, serverKeyPEM, err = p.storage.load()
+	shouldCreateNew := false
+	if err != nil || len(serverCertPEM) == 0 || len(serverKeyPEM) == 0 {
+		shouldCreateNew = true
+	} else {
+		certValid, checkErr := p.checker.checkExpiry(ctx, config.caCertPEM)
```
[the remark this answers is not in the document]
Comment (skhalash) [reply internal/webhookcert/server_cert_provider.go]: Done

File: main.go (hunk 1 above)
Comment (skhalash) [reply main.go]: Done

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**